infor-design/enterprise-wc
# 149 - Add Lookup Component and related Datagrid Features

**Explain the details for making this change. What existing problem does the pull request solve?**

Adds a MVD ids-lookup. And also some related datagrid features including

- List Style Datagrid
- Selection (single/mixed/multiple)
- Also some cleanup in trigger field to make things a bit easier
- Needed fixes to modal/popup for the lookup
- Some more test coverage

**Related github/jira issue (required)**:
Fixes #149 

**Steps necessary to review your pull request (required)**:

**Test Lookup**
- go to http://localhost:4300/ids-lookup
- click the first lookup and select from the list (results reflected in the field)
- Try to test the other examples in this page
- watch console for events

**Test List Style**
- go to http://localhost:4300/ids-data-grid/list-style.html
- check list style / test themes ect

**Test Selection Modes**
- go to http://localhost:4300/ids-data-grid/single-selection.html and test selection (note that this one has supressDeselection on which means you cant unselect a row)
- watch console for events
- go to http://localhost:4300/ids-data-grid/mixed-selection.html and test mixed select (not this means clicking the row activates and clicking the checkbox selects)
- watch console for events
- go to http://localhost:4300/ids-data-grid/multiple-selection.html and test multi select
- watch console for events

**Included in this Pull Request**:
